recipes-connectivity/network-manager-applet/network-manager-applet-1.4.4/openxt-menus.patch

```diff
@@ -0,0 +1,469 @@
++						G_CALLBACK (applet_device_disconnect_db),
++						info,
++						(GClosureNotify) applet_device_info_destroy, 0);
++			gtk_widget_set_sensitive (item, TRUE);
++		}
+ 		break;
+ 	}
+ 	default:
+@@ -1443,6 +1477,9 @@ nma_menu_add_vpn_submenu (GtkWidget *men
+ 	GPtrArray *list;
+ 	int i;
+ 
++	/* Don't show VPN menu */
++	return;
++
+ 	nma_menu_add_separator_item (menu);
+ 
+ 	vpn_menu = GTK_MENU (gtk_menu_new ());
+@@ -1537,30 +1574,6 @@ nma_set_networking_enabled_cb (GtkWidget
+ 	nm_client_networking_set_enabled (applet->nm_client, state, NULL);
+ }
+ 
+-
+-static void
+-nma_set_notifications_enabled_cb (GtkWidget *widget, NMApplet *applet)
+-{
+-	gboolean state;
+-
+-	g_return_if_fail (applet != NULL);
+-
+-	state = gtk_check_menu_item_get_active (GTK_CHECK_MENU_ITEM (widget));
+-
+-	g_settings_set_boolean (applet->gsettings,
+-	                        PREF_DISABLE_CONNECTED_NOTIFICATIONS,
+-	                        !state);
+-	g_settings_set_boolean (applet->gsettings,
+-	                        PREF_DISABLE_DISCONNECTED_NOTIFICATIONS,
+-	                        !state);
+-	g_settings_set_boolean (applet->gsettings,
+-	                        PREF_DISABLE_VPN_NOTIFICATIONS,
+-	                        !state);
+-	g_settings_set_boolean (applet->gsettings,
+-	                        PREF_SUPPRESS_WIFI_NETWORKS_AVAILABLE,
+-	                        !state);
+-}
+-
+ static gboolean
+ has_usable_wifi (NMApplet *applet)
+ {
+@@ -1606,19 +1619,22 @@ static void nma_menu_show_cb (GtkWidget
+ 
+ 	if (nm_client_get_state (applet->nm_client) == NM_STATE_ASLEEP) {
+ 		nma_menu_add_text_item (menu, _("Networking disabled"));
+-		return;
+-	}
++	} else {
+ 
+-	nma_menu_add_devices (menu, applet);
+-	nma_menu_add_vpn_submenu (menu, applet);
++		nma_menu_add_devices (menu, applet);
++		nma_menu_add_vpn_submenu (menu, applet);
+ 
+-	if (has_usable_wifi (applet)) {
+-		/* Add the "Hidden Wi-Fi network..." entry */
+-		nma_menu_add_separator_item (menu);
+-		nma_menu_add_hidden_network_item (menu, applet);
+-		nma_menu_add_create_network_item (menu, applet);
++		if (has_usable_wifi (applet)) {
++			/* Add the "Hidden Wi-Fi network..." entry */
++			nma_menu_add_separator_item (menu);
++			nma_menu_add_hidden_network_item (menu, applet);
++			nma_menu_add_create_network_item (menu, applet);
++		}
+ 	}
+ 
++	nma_context_menu_create (GTK_MENU_SHELL (menu), applet);
++	nma_context_menu_update (applet);
++
+ 	if (!INDICATOR_ENABLED (applet))
+ 		gtk_widget_show_all (menu);
+ }
+@@ -1793,15 +1809,12 @@ applet_connection_info_cb (NMApplet *app
+  * Generate the contextual popup menu.
+  *
+  */
+-static GtkWidget *nma_context_menu_create (NMApplet *applet)
++static void nma_context_menu_create (GtkMenuShell *menu, NMApplet *applet)
+ {
+-	GtkMenuShell *menu;
+ 	guint id;
+ 	static gboolean icons_shown = FALSE;
+ 
+-	g_return_val_if_fail (applet != NULL, NULL);
+-
+-	menu = GTK_MENU_SHELL (gtk_menu_new ());
++	g_assert (applet != NULL);
```

Comment:
The return type of the function is changed to void, keeping `g_return_val_if_fail` will result in error. In order to not proceed without checking `applet` being `NULL`, `g_assert` is introduced instead of `g_return_val_if_fail`